pytng/pytng.pyx

```diff
@@ -1016,334 +1138,340 @@ cdef class TNGFileIterator:
+                    self.read_step(i)
+                    yield self.current_integrator_step
             return sliceiter(start, stop, step)
         else:
             raise TypeError("Trajectories must be an indexed using an integer,"
                             " slice or list of indices")
 
 
 cdef class TNGCurrentIntegratorStep:
-    """Holds data blocks at the curent trajectory step"""
+    """Retrieves data at the curent trajectory step"""
 
     cdef bint debug
     cdef int64_t _n_blocks
-    cdef dict blocks  # TODO should we use a fixed size container?
 
-    def __cinit__(self, bint debug=False):
+    cdef tng_trajectory * _traj
+    cdef int64_t step
+
+    def __cinit__(self, TrajectoryWrapper traj, int64_t step,
+                  bint debug=False):
         self.debug = debug
-        self.blocks = {}
 
-    cdef inline void add_block(self, int64_t block_id, TNGDataBlock block):
-        self.blocks[block_id] = block  # add the block to the block dictionary
+        self._traj = traj._ptr
+        self.step = step
+
+    def __dealloc__(self):
+        pass
 
     @property
-    def block_set(self):
-        return self.blocks  # expose blocks to python layer
-
+    def step(self):
+        """The current integrator step being read
 
-cdef class TNGDataBlock:
-    """Contains the actual data in a tng_data_block
-       and exposes it as a Numpy array"""
+        Returns
+        -------
+        step : int
+            the current step in the TNG file
+        """
+        return self.step
 
-    cdef tng_trajectory * _traj
-    cdef int64_t _frame
-    cdef int64_t block_id
-    cdef bint debug
+    cpdef  get_time(self):
+        """Get the time of the current integrator step being read from the file
 
-    cdef int64_t step
-    cdef double frame_time
-    cdef double precision
-    cdef int64_t n_values_per_frame, n_atoms
-    cdef char[TNG_MAX_STR_LEN] block_name
-    cdef MemoryWrapper _wrapper  # manages the numpy array lifetime
-    cdef tng_function_status read_stat
-    cdef np.ndarray values  # the final values as a numpy array
+        Returns
+        -------
+        time : int
+            the time of the current step
+        """
+        cdef tng_function_status read_stat
+        cdef double _step_time
+        read_stat = self._get_step_time( & _step_time)
+        if read_stat != TNG_SUCCESS:
+            return None
+        else:
+            return _step_time
 
-    cdef bint block_is_read
+    cpdef np.ndarray get_positions(self, np.ndarray data):
+        """Get the positions present at the current step and read them into a
+        NumPy array
 
-    def __cinit__(self, TrajectoryWrapper traj, int64_t frame,
-                  bint debug=False):
+        Parameters
+        ----------
+        data : np.ndarray
+           NumPy array to read the data into. As this is a particle dependent
+           block, the shape should be (n_atoms, n_values_per_frame)
+           ie (n_atoms, 3).
```

Comment:
To clarify, when it says the shape should be `(n_atoms, n_values_per_frame)` is that different from `(n_atoms, ndim)`? I think when I was reading through the examples at the start I was under the impression this was the number of dims? Maybe it is meant to be the same thing.

Reply:
I think meant to be the same thing.

I think name `n_values_per_frame` (used in the library itself) is meant to provide the idea that for non-particle dependent blocks e.g box vectors we should expect that many values.

Open to changing if you think it would be clearer :)